test/Serilog.Tests/Settings/KeyValuePairSettingsTests.cs

```diff
@@ -152,5 +153,90 @@ public void TestMinimumLevelOverrides()
             Assert.NotNull(evt);
         }
 
+        [Theory]
+        [InlineData("Debug", LogEventLevel.Debug, "proper parsing")]
+        [InlineData("Information", LogEventLevel.Information, "proper parsing")]
+        [InlineData("", LogEventLevel.Information, "default value")]
+        public void LoggingLevelSwitchIsInstantiatedFromLogLevelAsString(string paramValue, LogEventLevel expectedSwitchInitialLevel, string reason)
+        {
+            object actual = null;
+            var didSucceed = KeyValuePairSettings.TryInstantiate(typeof(LoggingLevelSwitch), paramValue, out actual);
+
+            Assert.True(didSucceed, "TryInstantiate should successfully instantiate a LoggingLevelSwitch");
+            Assert.NotNull(actual);
+            Assert.IsType<LoggingLevelSwitch>(actual);
+            Assert.Equal(expectedSwitchInitialLevel, ((LoggingLevelSwitch)actual).MinimumLevel);
+        }
+
+        [Fact]
+        public void LoggingLevelSwitchIsConfigured()
+        {
+            var settings = new Dictionary<string, string>
+            {
+                ["level-switch:Switch1"] = "Information",
+                ["minimum-level:controlled-by"] = "Switch1",
```

Comment:
`<add key="serilog:level-switch:Switch1" value="Information" />` contains the same information but also enforces that if a switch is declared, we set its level. Separating the two means we can declare a switch without using it, or try to use a nonexistent switch. Do we gain anything by splitting the declaration and usage across two directives?

Reply:
Sorry, I'm not sure I understand your comment U_U 

I just based my implementation on your proposal from here : https://github.com/serilog/serilog-settings-appsettings/issues/7#issuecomment-325182308 

The switch needs to be used twice, both for the : 
- `.MinimumLevel.ControlledBy(LoggingLevelSwitch ls)`
- `.WriteTo.Seq( LoggingLevelSwitch controlLevelSwitch)`
... so we need to find a way to express that both parameters refer to the same instance of a LoggingSwitch , or did I miss something ?

Reply:
Sorry, yes, you're entirely correct, I misread the code in my attempt to skim through quickly.